README.md

```diff
@@ -114,13 +115,39 @@ an API-key and the resource type. To read from a table you should also specify t
 
 ### Writing
 
-To write to a resource you'll need to register a DataFrame that was read from
+There are two ways you can use cdp-spark-datasource to write to CDF: using `insertInto` or using the `save` function. 
+- `insertInto` - will check that all fields are present and in the correct order, and can be more convenient when
+working with Spark SQL tables.
+- `save` - will give you control over how to handle potential collisions with existing data,
+and allows updating a subset of fields in a row. 
+
+#### Writing with `.insertInto()`
+
+To write to a resource using the insert into pattern you'll need to register a DataFrame that was read from
 that resource as a temporary view. You'll need a project where you have write access
 and replace `myApiKey` in the examples below.
 
 Your schema will have to match that of the target exactly. A convenient way to ensure
 this is to copy the schema from the DataFrame you read into with `sourceDf.select(destinationDf.columns.map(col):_*)`, see time series example.
 
+`.insertInto()` will do upsert, as in updating existing rows and inserting new rows, for events, assets and time series.
```

Comment:
Typically on "id" for upsert, I guess?

Reply:
Yep, and source+sourceId for events. Coming to think of it assets actually don't have upserts on insertInto.